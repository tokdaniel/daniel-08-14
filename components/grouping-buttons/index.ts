export { default as GroupingButtons } from './grouping-buttons'
