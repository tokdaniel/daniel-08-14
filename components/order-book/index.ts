export { default as OrderBook } from './order-book'
export { default as HalfBook } from './half-book'
