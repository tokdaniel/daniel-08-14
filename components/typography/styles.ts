import { css } from 'styled-components'

const font = css`
  font-family: 'Lato', 'sans-serif';
`

const h1 = css`
  ${font};
  font-size: 80px;
  font-weight: 500;
`

const h2 = css`
  ${font};
  font-style: normal;
  font-weight: 300;
  font-size: 60px;
  line-height: 72.54px;
`

const h3 = css`
  ${font};
  font-style: normal;
  font-weight: 700;
  font-size: 40px;
  line-height: 48.84px;
`

const h4 = css`
  ${font};
  font-style: normal;
  font-weight: 700;
  font-size: 34px;
  line-height: 41.51px;
`

const h5 = css`
  ${font};
  font-style: normal;
  font-weight: 300;
  font-size: 26px;
  line-height: 35.41px;
  letter-spacing: 2.6px;
`

const h6 = css`
  ${font};
  font-style: normal;
  font-weight: 300;
  font-size: 16px;
  line-height: 35.41px;
  letter-spacing: 19.54px;
`

const subtitle1 = css`
  ${font};
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 19.54px;
`

const subtitle2 = css`
  ${font};
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  text-transform: uppercase;
  line-height: 17.09px;
  letter-spacing: 2px;
`

const body1 = css`
  ${font};
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
  letter-spacing: 2px;
`
const body2 = css`
  ${font};
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  text-transform: uppercase;
  line-height: 14.65px;
`

const body3 = css`
  ${font};
  font-style: normal;
  font-weight: 500;
  font-size: 10px;
  line-height: 12.21px;
`

const label = css`
  ${body3}
  text-transform: uppercase;
`

export const styles = {
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  'subtitle-1': subtitle1,
  'subtitle-2': subtitle2,
  'body-1': body1,
  'body-2': body2,
  'body-3': body3,
  label: label,
}

export type Variant = keyof typeof styles
export const variants = Object.keys(styles) as Variant[]
