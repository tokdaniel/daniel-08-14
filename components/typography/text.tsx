import styled, { CSSProperties } from 'styled-components'

import { Color } from 'config/theme.config'

import { styles, Variant } from './styles'

type Props = {
  variant: Variant
  textColor?: Color
  padding?: CSSProperties['padding']
  margin?: CSSProperties['margin']
}

export const Text = styled.div<Props>`
  ${({ variant }) => styles[variant]};
  color: ${({ textColor, theme }) => {
    if (!textColor) return theme.palette.grey[200]
    const [color, shade] = textColor

    return theme.palette[color][shade]
  }};
  margin: ${({ margin }) => (margin ? margin : 'initial')};
  padding: ${({ padding }) => (padding ? padding : 'initial')};
`

export default Text
