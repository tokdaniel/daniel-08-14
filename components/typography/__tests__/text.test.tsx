import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'

import { Text } from 'components/typography'
import theme from 'config/theme.config'

describe('Text', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Text variant="h1" />
          <Text variant="h2" />
          <Text variant="h3" />
          <Text variant="h4" />
          <Text variant="h5" />
          <Text variant="h6" />
          <Text variant="body-1" />
          <Text variant="body-2" />
          <Text variant="body-3" />
          <Text variant="subtitle-1" />
          <Text variant="subtitle-2" />
          <Text variant="label" />
        </ThemeProvider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
