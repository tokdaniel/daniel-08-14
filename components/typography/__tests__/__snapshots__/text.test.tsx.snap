// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
Array [
  <div
    className="sc-bdnxRM dgVgsH"
  />,
  <div
    className="sc-bdnxRM knLCNz"
  />,
  <div
    className="sc-bdnxRM gFJuKx"
  />,
  <div
    className="sc-bdnxRM lfHdrH"
  />,
  <div
    className="sc-bdnxRM ddgugd"
  />,
  <div
    className="sc-bdnxRM ejWiHb"
  />,
  <div
    className="sc-bdnxRM cLcjEa"
  />,
  <div
    className="sc-bdnxRM kyahsn"
  />,
  <div
    className="sc-bdnxRM hhOKzm"
  />,
  <div
    className="sc-bdnxRM fmZhhZ"
  />,
  <div
    className="sc-bdnxRM gJuksH"
  />,
  <div
    className="sc-bdnxRM jDkucz"
  />,
]
`;
